### PR TITLE
feat(rxNav): check for oriLocationService and use it in search

### PR DESCRIPTION
**Pull request by halkon**
JIRA: https://jira.rax.io/browse/FRMW-1035

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] QE LGTM

### CODE REVIEW

File: src/components/rxApp/scripts/rxBillingSearch.js [hunk 1]
```diff
@@ -18,10 +18,13 @@ angular.module('encore.ui.rxApp')
...
-                        if (isBilling) {
-                            $location.url('/search?q=' + searchValue + '&type=' + scope.searchType);
+                        var path = '/search?q=' + searchValue + '&type=' + scope.searchType;
+                        if ($injector.has('oriLocationService')) {
+                            $injector.get('oriLocationService').setCanvasURL('/billing' + path);
+                        } else if (isBilling) {
+                            $location.url(path);
                         } else {
-                            $window.location = '/billing/search?q=' + searchValue + '&type=' + scope.searchType;
+                            $window.location = '/billing' + path;
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxBillingSearch.js]: Is this ok to change. It looks like it'd change existing functionality.

Comment (halkon) [inline src/components/rxApp/scripts/rxBillingSearch.js]: Yeah I just moved the URL construction to the variable above.

Comment (CITguy) [inline src/components/rxApp/scripts/rxBillingSearch.js]: ah, I guess I should've read the changes first :)

File: src/components/rxApp/scripts/rxAccountSearch.js [hunk 2]
```diff
@@ -5,14 +5,19 @@ angular.module('encore.ui.rxApp')
...
-.directive('rxAccountSearch', function ($window) {
+.directive('rxAccountSearch', function ($window, $injector) {
     return {
         templateUrl: 'templates/rxAccountSearch.html',
         restrict: 'E',
         link: function (scope) {
             scope.fetchAccount = function (searchValue) {
                 if (!_.isEmpty(searchValue)) {
-                    $window.location = '/search?term=' + searchValue;
+                    var path = '/search?term=' + searchValue;
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxAccountSearch.js]: We can use an ES6 template literal here.

```javascript
var path = `/search?term=${searchValue}`;
```

File: src/components/rxApp/scripts/rxAtlasSearch.js [hunk 3]
```diff
@@ -6,14 +6,19 @@ angular.module('encore.ui.rxApp')
...
-.directive('rxAtlasSearch', function ($window) {
+.directive('rxAtlasSearch', function ($window, $injector) {
     return {
         template: '<rx-app-search placeholder="Search by username..." submit="searchAccounts"></rx-app-search>',
         restrict: 'E',
         link: function (scope) {
             scope.searchAccounts = function (searchValue) {
                 if (!_.isEmpty(searchValue)) {
-                    $window.location = '/cloud/' + searchValue + '/servers/';
+                    var path = '/cloud/' + searchValue + '/servers/';
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxAtlasSearch.js]: We can use an ES6 template literal here:

```javascript
var path = `/cloud/${searchValue}/servers/`;
```

File: src/components/rxApp/scripts/rxBillingSearch.js [hunk 4]
```diff
@@ -18,10 +18,13 @@ angular.module('encore.ui.rxApp')
                 if (!_.isEmpty(searchValue)) {
                     // Assuming we are already in /billing, we should use $location to prevent a page refresh
                     encoreRoutes.isActiveByKey('billing').then(function (isBilling) {
-                        if (isBilling) {
-                            $location.url('/search?q=' + searchValue + '&type=' + scope.searchType);
+                        var path = '/search?q=' + searchValue + '&type=' + scope.searchType;
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxBillingSearch.js]: We can use an ES6 template literal here:

```javascript
var path = `/search?q=${searchValue}&type=${scope.searchType}`;
```

**Comment by CITguy** [comment]
Dev LGTM

**Comment by halkon** [comment]
@CITguy apparently browsers support it, but not our tooling...
https://travis-ci.org/rackerlabs/encore-ui/builds/161096159#L2041-L2043
```
PhantomJS 1.9.8 (Linux) ERROR
  SyntaxError: Parse error
  at /home/travis/build/rackerlabs/encore-ui/src/components/rxApp/scripts/rxAccountSearch.js:9
```
Maybe we finally need to upgrade to `PhantomJS 2`?

**Comment by CITguy** [comment]
oh come on

**Comment by CITguy** [comment]
@halkon: okay, you can put it back the way it was. Upgrading phantomJS right now is out of scope. 😢  🐼

**Comment by CITguy** [comment]
Still have my LGTM, though.

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**